storybookjs/storybook
# NextJS: Fix default next image loader when src has params

Closes #24104






## What I did

This PR fixes a bug in `@storybook/nextjs` which malforms URLs with search parameters in them given to `next/image`.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

Couldn't find any existing tests to add to.


#### Manual testing

1. Create a new NextJS storybook
2. Add a component that uses `next/image` with a url that specifies query parameters
3. Observe that the image loads because the URL is no longer malformed

### Documentation



No documentation updates were needed.

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
